ddtrace/tracer/sqlcomment.go

```diff
@@ -188,7 +190,56 @@ func commentQuery(query string, tags map[string]string) string {
 	return b.String()
 }
 
-// Extract is not implemented on SQLCommentCarrier
+// Extract parses for key value attributes in a sql query's comment in order to build a span context
 func (c *SQLCommentCarrier) Extract() (ddtrace.SpanContext, error) {
-	return nil, nil
+	var ctx spanContext
+	re := regexp.MustCompile(`/\*(.*?)\*/`) // extract sql comment
+	if match := re.FindStringSubmatch(c.Query); len(match) == 2 {
```

Comment:
The datadog tracer prepends the sqlcomment to the query but the open-telemetry implementation _appends_ it to the end of the query. I think it might make sense to try and find a comment with `traceparent` regardless of its position in the query text.